src/App.tsx

```diff
@@ -15,47 +16,123 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+// Use this function in the render method to prepare goods
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  // To avoid the original array mutation
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort((good1, good2) => {
+    switch (sortType) {
+      case SortType.ALPHABET:
+        return good1.localeCompare(good2);
+
+      case SortType.LENGTH:
+        return good1.length - good2.length;
+
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
+// DON'T save goods to the state
+// type State = {
+//   isReversed: boolean,
+//   sortType: SortType,
+// };
+
 export const App: React.FC = () => {
+  const [isReversed, setReverse] = useState(false);
+  const [sortType, setSortType] = useState(SortType.NONE);
+
+  const sort = (getSortType: SortType) => {
+    setSortType(getSortType);
+  };
+
+  const reverse = () => {
+    setReverse(!isReversed);
+  };
+
+  const reset = () => {
+    setReverse(false);
+    setSortType(SortType.NONE);
+  };
+
+  const isResetVisible = sortType !== SortType.NONE || isReversed;
+  const visibleGoods = getReorderedGoods(
+    goodsFromServer,
+    { sortType, isReversed },
+  );
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={cn('button is-info', {
+            'is-light': sortType !== SortType.ALPHABET,
+          })}
+          onClick={() => sort(SortType.ALPHABET)}
```

Comment:
no need of additional wrapper here
```suggestion
          onClick={() => setSortType(SortType.ALPHABET)}
```